contracts/CompoundGovernor.sol

```diff
@@ -33,6 +33,54 @@ contract CompoundGovernor is
     GovernorSettableFixedQuorumUpgradeable,
     OwnableUpgradeable
 {
+    /// @notice Emitted when the expiration of a whitelisted account is set or updated.
+    /// @param account The address of the account being whitelisted.
+    /// @param expiration The timestamp until which the account is whitelisted.
+    event WhitelistAccountExpirationSet(address account, uint256 expiration);
+
+    /// @notice Emitted when the whitelistGuardian is set or changed.
+    /// @param oldGuardian The address of the previous whitelistGuardian.
+    /// @param newGuardian The address of the new whitelistGuardian.
+    event WhitelistGuardianSet(address oldGuardian, address newGuardian);
+
+    /// @notice Emitted when the proposalGuardian is set
+    /// @notice Emitted when the proposal guardian is set or updated.
+    /// @param oldProposalGuardian The address of the previous proposal guardian.
+    /// @param oldProposalGuardianExpiry The expiration timestamp of the previous proposal guardian's role.
+    /// @param newProposalGuardian The address of the new proposal guardian.
+    /// @param newProposalGuardianExpiry The expiration timestamp of the new proposal guardian's role.
+    event ProposalGuardianSet(
+        address oldProposalGuardian,
+        uint96 oldProposalGuardianExpiry,
+        address newProposalGuardian,
+        uint96 newProposalGuardianExpiry
+    );
+
+    /// @notice Error thrown when an unauthorized address attempts to perform a restricted action.
+    /// @param reason A brief description of why the caller is unauthorized.
+    /// @param caller The address that attempted the unauthorized action.
+    error Unauthorized(bytes32 reason, address caller);
+
+    /// @notice Address which manages whitelisted proposals and whitelist accounts.
+    /// @dev This address has the ability to set account whitelist expirations and can be changed through the governance
+    /// process.
+    address public whitelistGuardian;
+
+    /// @notice Account which has the ability to cancel proposals. This privilege expires at the given expiration
+    /// timestamp.
+    ProposalGuardian public proposalGuardian;
+
+    /// @notice Stores the expiration of account whitelist status as a timestamp
+    mapping(address account => uint256 timestamp) public whitelistAccountExpirations;
+
+    /// @notice The address and expiration of the proposal guardian.
+    struct ProposalGuardian {
```

Comment:
I'm guessing the reason bravo governor uses `ProposalGuardian` as a struct instead of a mapping is because we only have one `proposalGuardian`, and this is more gas efficient. Is that understanding correct?